cou/steps/plan.py

```diff
@@ -64,18 +64,23 @@ async def generate_plan(analysis_result: Analysis, backup_database: bool) -> Upg
     :raises OutOfSupportRange: When the OpenStack release or Ubuntu series is out of the current
     supporting range.
     :return: Plan with all upgrade steps necessary based on the Analysis.
-    :rtype: UpgradeStep
+    :rtype: BaseStep
     """
     target = determine_upgrade_target(
         analysis_result.current_cloud_os_release, analysis_result.current_cloud_series
     )
-    print(f"Upgrading cloud from '{analysis_result.current_cloud_os_release}' to '{target}'\n.")
 
-    plan = UpgradeStep(description="Top level plan", parallel=False)
+    plan = BaseStep(
```

Comment:
I think that `BaseStep` should follow some abstraction, so it should be defined as abstract class and like that it should never be used directly.
This should be defined as `UpgradePlan(description=...)`, where `UpgradePlan` will set `parallel` to `False` and `coro` to `None`. I think that description should be only parameter for `UpgradePlan`, because it does not make sense to run in parallel or with any coroutine.